### PR TITLE
chore(docs): a custom theme guide

### PR DESCRIPTION
**Pull request by ushkal**

### CODE REVIEW

File: packages/dx-react-grid/docs/guides/custom-theme.md [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+
+## Implementing a custom theme
+For example let's make a grid theme based on the [Semantic UI](https://react.semantic-ui.com/) framework. This theme will cover three essential plugins - [Grid](https://devexpress.github.io/devextreme-reactive/react/grid/docs/reference/grid/), [Table](https://devexpress.github.io/devextreme-reactive/react/grid/docs/reference/table/) and [TableHeader](https://devexpress.github.io/devextreme-reactive/react/grid/docs/reference/table-header-row/). Theme sources and preview are available in this [CodeSandbox](https://codesandbox.io/s/jmqwvjqw3).
+
+### Theme structure
+Create a new directory in the sources root called `semantic-ui-theme` with two nested directories - `plugins` and `templates`.
```
Comment (MaximKudriavtsev) [inline packages/dx-react-grid/docs/guides/custom-theme.md]: Maybe draw something like that?
```
- semantic-ui-package
--- plugins
--- templates
```

File: packages/dx-react-grid/docs/guides/custom-theme.md [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+```jsx
+import { TableHeaderRow as TableHeaderRowBase } from '@devexpress/dx-react-grid';
+import { TableRow } from '../templates/table-row';
+// ... other components
+
+export const TableHeaderRow = props => (
+  <TableHeaderRowBase
+    rowComponent={TableRow}
+    // ... other components
+    {...props}
```
Comment (MaximKudriavtsev) [inline packages/dx-react-grid/docs/guides/custom-theme.md]: Can we add the end of a piece of code? Because this looks bad(

Comment (ushkal) [reply packages/dx-react-grid/docs/guides/custom-theme.md]: Whole piece of code will take a lot of space

**Comment by SergeyAlexeev** [comment]
> https://codesandbox.io/s/jmqwvjqw3

*templates/sort-label.jsx:*

```jsx
export const SortLabel = ({ ...restProps }) => <span {...restProps} />;
```

->

```jsx
export const SortLabel = props => <span {...props} />;
```

File: packages/dx-react-common/docs/guides/custom-theme.md [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+
+#### Handling props
+Sometimes theme component may be simple like
+```jsx
+import { Table as TableSUI } from 'semantic-ui-react';
+const TableHead = props => <TableSUI.Header {...props} />;
+```
+Here a component passes props as-is to the underlying Semantic UI component.
+
+But there are also other templates that receive grid-specific data via props and a different approach is required. These properties are described in a related plugin docs under section 'Interfaces'. Let's take a `tableCell` component for example. It receives props of type [Table.DataCellProps](https://devexpress.github.io/devextreme-reactive/react/grid/docs/reference/table/#tabledatacellprops). As you can see, some of these properties - such as `tableColumn`, `row`, `value` - are unknown to React, hence we should destructure the props.
```
Comment (SergeyAlexeev) [inline packages/dx-react-common/docs/guides/custom-theme.md]: > But there are also other templates...

I'd prefer 'But there are also other **components**...'

File: packages/dx-react-grid/docs/guides/custom-theme.md (hunk 2 above)
Comment (ushkal) [reply packages/dx-react-grid/docs/guides/custom-theme.md]: Yeah that's a good idea

File: packages/dx-react-common/docs/guides/custom-theme.md [hunk 4]
```diff
@@ -0,0 +1,76 @@
+# React Components - Create a Custom Theme
+
+## Motivation
+[React Components](https://devexpress.github.io/devextreme-reactive/react/) are shipped with modern themes and can be easily customized. However, there are situations when you need to develop your own theme, for example, when the design guidelines of your site go against the Bootstrap or Material-UI guidelines.
```
Comment (dirk-pieterse) [inline packages/dx-react-common/docs/guides/custom-theme.md]: modern themes? can be easily customized?

Comment (dirk-pieterse) [reply packages/dx-react-common/docs/guides/custom-theme.md]: BTW, why is there a 'Motivation' section?

File: packages/dx-react-common/docs/guides/custom-theme.md [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+
+## How Themes Work
+A theme contains only the representation logic. All the internal routines are encapsulated in plugins and contained in core packages: `dx-react-grid`, `dx-react-scheduler`, and `dx-react-chart`. A core package exports "bare" UI plugins that a theme package can fill with visual components via [Render Props](https://reactjs.org/docs/render-props.html). Essentially, a theme is a set of React UI components delivered to plugins. To stylize a plugin, you just need to provide necessary components to it.
```
Comment (dirk-pieterse) [inline packages/dx-react-common/docs/guides/custom-theme.md]: Essentially?

Comment (dirk-pieterse) [reply packages/dx-react-common/docs/guides/custom-theme.md]: delivered to?

Comment (dirk-pieterse) [reply packages/dx-react-common/docs/guides/custom-theme.md]: you just need?

File: packages/dx-react-common/docs/guides/custom-theme.md [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+
+## Implement a Custom Theme
+Let's create a grid theme based on the [Semantic UI](https://react.semantic-ui.com/) framework. The theme will stylize three main plugins: [Grid](https://devexpress.github.io/devextreme-reactive/react/grid/docs/reference/grid/), [Table](https://devexpress.github.io/devextreme-reactive/react/grid/docs/reference/table/), and [TableHeader](https://devexpress.github.io/devextreme-reactive/react/grid/docs/reference/table-header-row/). Theme sources and preview are available in this [CodeSandbox](https://codesandbox.io/s/jmqwvjqw3).
```
Comment (dirk-pieterse) [inline packages/dx-react-common/docs/guides/custom-theme.md]: Let's create?

**Pull request merged**